library/src/amd_detail/hipsparse.cpp

```diff
@@ -14624,6 +14624,53 @@ hipsparseStatus_t hipsparseSpGEMM_workEstimation(hipsparseHandle_t          hand
     return HIPSPARSE_STATUS_SUCCESS;
 }
 
+hipsparseStatus_t hipsparseSpGEMM_estimateMemory(hipsparseHandle_t          handle,
+                                                 hipsparseOperation_t       opA,
+                                                 hipsparseOperation_t       opB,
+                                                 const void*                alpha,
+                                                 hipsparseConstSpMatDescr_t matA,
+                                                 hipsparseConstSpMatDescr_t matB,
+                                                 const void*                beta,
+                                                 hipsparseSpMatDescr_t      matC,
+                                                 hipDataType                computeType,
+                                                 hipsparseSpGEMMAlg_t       alg,
+                                                 hipsparseSpGEMMDescr_t     spgemmDescr,
+                                                 float                      chunk_fraction,
+                                                 size_t*                    bufferSize3,
+                                                 void*                      externalBuffer3,
+                                                 size_t*                    bufferSize2)
+{
+    if(handle == nullptr || alpha == nullptr || beta == nullptr || matA == nullptr
+       || matB == nullptr || matC == nullptr || bufferSize3 == nullptr)
+    {
+        return HIPSPARSE_STATUS_INVALID_VALUE;
+    }
+
+    if(externalBuffer3 == nullptr)
+    {
+        *bufferSize3 = 4;
```

Comment:
Why not put 0?

Reply:
If I put 0 then the externalBuffer3 will remain nullptr as nothing will be allocated and therefore we will never enter the else branch.

Reply:
mmhm I see. I do not really like that.. but I guess we probably have to live with it.